packages/next/lib/is-serializable-props.ts

```diff
@@ -0,0 +1,138 @@
+const regexpPlainIdentifier = /^[A-Za-z_$][A-Za-z0-9_$]*$/
+
+function isPlainObject(value: any): boolean {
+  if (Object.prototype.toString.call(value) !== '[object Object]') {
+    return false
+  }
+
+  const prototype = Object.getPrototypeOf(value)
+  return prototype === null || prototype === Object.prototype
```

Comment:
Should we support calling `.toJSON()` on objects since it `JSON.stringify()` [handles this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)? e.g. `JSON.stringify({ a: new Date() })` returns `'{"a":"2020-03-09T17:17:44.849Z"}'`